### PR TITLE
extract `IconPath` type for uri, uris, and ThemeIcon combo

### PR DESCRIPTION
**Pull request by jrieken**
re https://github.com/microsoft/vscode/issues/124363

### CODE REVIEW

File: src/vscode-dts/vscode.d.ts [hunk 1]
```diff
@@ -946,6 +946,21 @@ declare module 'vscode' {
 		constructor(id: string, color?: ThemeColor);
 	}
 
+	/**
+	 * Represents an icon in the UI. This is either an uri, separate uris for the light- and dark-themes,
+	 * or a {@link ThemeIcon theme icon}.
+	 */
+	export type IconPath = Uri | {
```
Comment (jrieken) [inline src/vscode-dts/vscode.d.ts:953]: @mjbvz Let me know if you have an idea for a better name. This isn't overly correct but matches the property names. 

Also there are still a few places where `iconPath: ...` is used/defined but where not this exact shape is supported

**Review by mjbvz — APPROVED** [review]
lgtm. I think `IconPath` makes sense with the property names

File: src/vscode-dts/vscode.d.ts [hunk 2]
```diff
@@ -11902,16 +11899,7 @@ declare module 'vscode' {
...
-			/**
-			 * The icon path for the light theme.
-			 */
-			light: string | Uri;
-			/**
-			 * The icon path for the dark theme.
-			 */
-			dark: string | Uri;
-		} | ThemeIcon;
+		iconPath?: IconPath;
```
Comment (eamodio) [inline src/vscode-dts/vscode.d.ts:11902]: @jrieken since this type took a `string`, in addition to a `Uri`, isn't this a breaking change?

Comment (jrieken) [reply src/vscode-dts/vscode.d.ts:11902]: Yeah, sorry. That was an oversight

**Pull request merged**